gpt_index/indices/postprocessor/node.py

```diff
@@ -82,3 +90,229 @@ def postprocess_nodes(
                 new_nodes.append(node)
 
         return new_nodes
+
+
+class BasePrevNextNodePostProcessor(BaseNodePostprocessor):
+    """Previous/Next Node post-processor.
+
+    Allows users to fetch additional nodes from the document store,
+    based on the relationships of the nodes.
+
+    NOTE: this is a beta feature.
+
+    Args:
+        docstore (DocumentStore): The document store.
+        num_nodes (int): The number of nodes to return (default: 1)
+        mode (str): The mode of the post-processor. Can be "forward" or "backward".
+
+    """
+
+    docstore: DocumentStore
+    num_nodes: int = Field(default=1)
+
+    def _get_forward_nodes(self, node: Node) -> Dict[str, Node]:
```

Comment:
related to above: these don't necessarily need to be instance methods attached to a superclass.

Reply:
They can be standalone methods in the module, with docstore and num_modes arguments passed in.